File: app/src/main/java/com/karol/sezonnazdrowie/data/CalendarDayConverter.java
package com.karol.sezonnazdrowie.data;

import androidx.room.TypeConverter;

import com.prolificinteractive.materialcalendarview.CalendarDay;

import java.util.Calendar;

public class CalendarDayConverter {
    @TypeConverter
    public static CalendarDay fromTimestamp(Long value) {
        Calendar cal = Calendar.getInstance();
        if (value == null) {
            return null;
        }
        cal.setTimeInMillis(value);
        return CalendarDay.from(cal);
    }

    @TypeConverter
    public static Long dateToTimestamp(CalendarDay date) {
        return date == null ? null : date.getCalendar().getTimeInMillis();
    }
}

File: app/src/main/java/com/karol/sezonnazdrowie/data/CalendarDayConverter.kt
package com.karol.sezonnazdrowie.data

import androidx.room.TypeConverter

import com.prolificinteractive.materialcalendarview.CalendarDay

import java.util.Calendar

class CalendarDayConverter {

    @TypeConverter
    fun fromTimestamp(value: Long?): CalendarDay? {
        val cal = Calendar.getInstance()
        return if (value == null) {
            null
        } else {
            cal.timeInMillis = value
            CalendarDay.from(cal)
        }
    }

    @TypeConverter
    fun dateToTimestamp(date: CalendarDay?): Long? = date?.calendar?.timeInMillis
}
